src/schema.ts

```diff
@@ -63,15 +73,10 @@ const resolvers = {
       })
       return users
     },
-    movies: (parent: any, args: any, ctx: Context) => {
-      const filteredMovies = ctx.prisma.movie.findMany({
-        where: {
-          OR: [
-            { title: { contains: args.search } }
-          ]
-        }
-      })
-      return filteredMovies
+    movies: async (parent: any, args: any, ctx: Context) => {
+      let search = `%${args.search}%`
+      const movies = await ctx.prisma.raw(`SELECT * FROM public.movie WHERE LOWER(title) LIKE LOWER('${search}');`)
```

Comment:
The idea is good, but when using raw queries we are losing the protection against sql injection.
There are a few options usually to solve this:
- using prepared statements (preferred)
- using an escape function
I am not sure prisma2 has some function that does this, I could find anything so far. Maybe you can have a look, otherwise we will have to use another dependency for this.

Another thing, the '%' is part of the query not of the argument. If you would use search as an argument in a prepared statement it would search for titles that contain the % instead of treating them as a wildcard


```suggestion
      const search = some_escape_function(args.search)
      const movies = await ctx.prisma.raw(`SELECT * FROM public.movie WHERE LOWER(title) LIKE LOWER('%${search}%');`)
```

Reply:
I have searched for solutions with prisma2, but could not find anything useful.